tests/tensor/test_subtensor.py

```diff
@@ -105,11 +112,63 @@ def test_as_index_literal():
 
 
 class TestGetCanonicalFormSlice:
+    def test_none_raises(self):
+        with pytest.raises(ValueError, match="Can't create"):
+            get_canonical_form_slice(NoneConst, 5)
+        with pytest.raises(ValueError, match="Can't create"):
+            get_canonical_form_slice(None, 5)
+
     def test_scalar_constant(self):
         a = as_scalar(0)
         length = lscalar()
         res = get_canonical_form_slice(a, length)
-        assert res[0].owner.op == ptb.switch
+        assert isinstance(res[0].owner.op, ptb.ScalarFromTensor)
+        assert res[1] == 1
+
+    def test_tensor_constant(self):
+        a = as_tensor(0)
+        length = lscalar()
+        res = get_canonical_form_slice(a, length)
+        assert isinstance(res[0].owner.op, ptb.ScalarFromTensor)
+        assert res[1] == 1
+
+    def test_symbolic_scalar(self):
+        a = int16()
+        length = lscalar()
+        res = get_canonical_form_slice(a, length)
+        assert res[0].owner.op, ptb.switch
+        assert res[1] == 1
+
+    def test_symbolic_tensor(self):
+        a = lscalar()
+        length = lscalar()
+        res = get_canonical_form_slice(a, length)
+        assert isinstance(res[0].owner.op, ptb.ScalarFromTensor)
+        assert res[1] == 1
+
+    def test_constant_slice(self):
+        idx = as_symbolic_slice(slice(3, 7, 2))
+        assert isinstance(idx, SliceConstant)
```

Comment:
I don't think we need to support non python slices. The code before wasn't, so there is no specified need?

Reply:
Why are there symbolic slices to begin with?

If we don't need them, we should remove them. If we need them, I think the functions dealing with slices should deal with them too.

Reply:
Replied here: https://github.com/pymc-devs/pytensor/pull/823#discussion_r1650193758

They are used in a different class of Ops